frontend/server/src/Controllers/Run.php

```diff
@@ -1300,13 +1304,15 @@ private static function getGraderResourcePassthru(
     /**
      * Given the run resouce path, fetches its contents from S3.
      *
-     * @param  string $resourcePath The run's resource path.
-     * @param  bool   $passthru     Whether to output directly.
-     * @return ?string              The contents of the resource (or an empty string) if successful. null otherwise.
+     * @param  string       $resourcePath The run's resource path.
+     * @param  bool         $passthru     Whether to output directly.
+     * @param  list<string> $fileHeaders
+     * @return ?string                    The contents of the resource (or an empty string) if successful. null otherwise.
      */
     private static function downloadResourceFromS3(
         string $resourcePath,
-        bool $passthru
+        bool $passthru,
+        array $fileHeaders = []
```

Comment:
nit:

```suggestion
        array $headers = []
```

estas son cabeceras HTTP, no cabeceras del archivo.

Reply:
La voy a renombrar por `$httpHeaders`, debido a que ya hay una variable que se llama `$headers` en ese método.